src/common/cockpitwebresponse.c

```diff
@@ -985,6 +985,17 @@ substitute_message (const gchar *variable,
   return NULL;
 }
 
+static GBytes *
+substitute_hash_value (const gchar *variable,
+                       gpointer user_data)
+{
+  GHashTable *data = user_data;
+  gchar *value = g_hash_table_lookup (data, variable);
+  if (value)
+    return g_bytes_new (value, strlen (value));
+  return NULL;
```

Comment:
Do we want to replace with a zero length string when not present? If not, why not?